src/lib.rs

```diff
@@ -737,7 +737,7 @@ static USER_AGENT: &str = concat!(env!("CARGO_PKG_NAME"), " (", env!("CARGO_PKG_
 
 fn validate_package_and_version(package: &str, version: &str) -> Result<(), ApiError> {
     lazy_static! {
-        static ref PACKAGE_PATTERN: Regex = Regex::new(r#"^[a-z_-]+$"#).unwrap();
+        static ref PACKAGE_PATTERN: Regex = Regex::new(r#"^[a-z]\w*$"#).unwrap();
```

Comment:
1. Do we also validate the minlength:
`https://github.com/hexpm/hexpm/blob/d6bfdb1ecae7f05b826b252be2e53f6a59da7a38/lib/hexpm/repository/package.ex#L59C1-L59C1`
2. Could we make it simple and require gleam apps/libs created via `gleam new`?
3. Would hex.pm accept `a_` aka a trailing underscore, should we forbid trailing with a _ or - (`\w - matches any word character (equivalent to [a-zA-Z0-9_])`)

Reply:
1. Don't believe so, I suppose we could just replace the `*` with `+` in the regex to enforce this
2. I'm not sure what you mean here
3. I've never done elixir but looking at the code around the regex (and without trying it), I would guess it would be allowed?

As fair as I can tell, the requirements are:
- Min length 2
- Match regex `^[a-z]\w*$`
- Not be one of the reserved names (https://github.com/hexpm/hexpm/blob/d6bfdb1ecae7f05b826b252be2e53f6a59da7a38/lib/hexpm/repository/package.ex#L23-L48)
- Not be taken already